### PR TITLE
Te 1500 smb shadow copy

### PR DESCRIPTION
**Pull request by markbainix**
adding Shadow Copy test case
image

### CODE REVIEW

File: keywords/webui/smb.py [hunk 1]
```diff
@@ -19,6 +19,52 @@ def add_smb_acl_entry_by_api(cls, who: str, userid: int, permission: str, perm_t
...
+        Example:
+        - SMB.assert_file_exists('myFile', 'myShare', 'user', 'password')
+        - SMB.assert_file_exists('myFile', 'myShare', 'user', 'password', True)
+        """
+        use_ad = ""
+        if ad is True:
+            use_ad = "-W AD03 "
+        response = SSH_Command_Line(f'smbclient //{private_config["IP"]}/{share} {use_ad}-U {user}%{password} -c \'ls\'', private_config['SMB_ACL_IP'], private_config['SMB_USERNAME'], private_config['SMB_PASSWORD'])
+        return response.stdout.__contains__(file)
+
```
Comment (ericbsd) [inline keywords/webui/smb.py:67]: Should have these ssh keywords in `keywords/ssh/smb.py` or  `keywords/ssh/smbclient.py`

File: keywords/webui/smb.py [hunk 2]
```diff
@@ -102,6 +148,30 @@ def assert_user_can_access(cls, share: str, user: str, password: str) -> bool:
...
+        - SMB.assert_user_can_delete_directory('myDir', 'myShare', 'user', 'password')
+        - SMB.assert_user_can_delete_directory('myDir', 'myShare', 'user', 'password', True)
+        """
+        use_ad = ""
+        if ad is True:
+            use_ad = "-W AD03 "
+        SSH_Command_Line(f'smbclient //{private_config["IP"]}/{share} {use_ad}-U {user}%{password} -c \'rmdir {directory}\'', private_config['SMB_ACL_IP'], private_config['SMB_USERNAME'], private_config['SMB_PASSWORD'])
+        response = SSH_Command_Line(f'smbclient //{private_config["IP"]}/{share} {use_ad}-U {user}%{password} -c \'ls\'', private_config['SMB_ACL_IP'], private_config['SMB_USERNAME'], private_config['SMB_PASSWORD'])
+        return not response.stdout.__contains__(directory)
+
```
Comment (ericbsd) [inline keywords/webui/smb.py:174]: same here. It should be in `keywords/ssh/smb.py` or `keywords/ssh/smbclient,py`.

File: keywords/webui/smb.py [hunk 3]
```diff
@@ -116,15 +186,40 @@ def assert_user_can_delete_file(cls, file: str, share: str, user: str, password:
...
+        - SMB.assert_user_can_put_directory('myDir', 'myShare', 'user', 'password')
+        - SMB.assert_user_can_put_directory('myDir', 'myShare', 'user', 'password', True)
+        """
+        use_ad = ""
+        if ad is True:
+            use_ad = "-W AD03 "
+        SSH_Command_Line(f'cd smbdirectory; smbclient //{private_config["IP"]}/{share} {use_ad}-U {user}%{password} -c \'mkdir {directory}\'', private_config['SMB_ACL_IP'], private_config['SMB_USERNAME'], private_config['SMB_PASSWORD'])
+        response = SSH_Command_Line(f'cd smbdirectory; smbclient //{private_config["IP"]}/{share} {use_ad}-U {user}%{password} -c \'ls\'', private_config['SMB_ACL_IP'], private_config['SMB_USERNAME'], private_config['SMB_PASSWORD'])
+        return response.stdout.__contains__(directory)
+
```
Comment (ericbsd) [inline keywords/webui/smb.py:222]: Same here. It should be in `keywords/ssh/smb.py` or `keywords/ssh/smbclient,py`.

**Review by ericbsd — CHANGES_REQUESTED** [review]

File: keywords/webui/smb.py (hunk 1 above)
Comment (markbainix) [reply keywords/webui/smb.py:67]: probably. will move them in next PR

**Review by ericbsd — APPROVED** [review]

**Pull request merged**